app/core/Tracker2/SegmentTracker.js

```diff
@@ -0,0 +1,166 @@
+    'ready',
+    'alias',
+    'debug',
+    'page',
+    'once',
+    'off',
+    'on'
+  ];
+
+  // Define a factory to create stubs. These are placeholders
+  // for methods in Analytics.js so that you never have to wait
+  // for it to load to actually record data. The `method` is
+  // stored as the first argument, so we can replay the data.
+  analytics.factory = function(method){
+    return function(){
+      var args = Array.prototype.slice.call(arguments);
+      args.unshift(method);
+      analytics.push(args);
+      return analytics;
+    };
+  };
+
+  // For each of our methods, generate a queueing stub.
+  for (var i = 0; i < analytics.methods.length; i++) {
+    var key = analytics.methods[i];
+    analytics[key] = analytics.factory(key);
+  }
+
+  // Define a method to load Analytics.js from our CDN,
+  // and that will be sure to only ever load it once.
+  analytics.load = function(key, options){
+    // Create an async script element based on your key.
+    var script = document.createElement('script');
+    script.type = 'text/javascript';
+    script.async = true;
+    script.src = 'https://cdn.segment.com/analytics.js/v1/'
+      + key + '/analytics.min.js';
+
+    // Insert our script next to the first script element.
+    var first = document.getElementsByTagName('script')[0];
+    first.parentNode.insertBefore(script, first);
+    analytics._loadOptions = options;
+  };
+
+  // Add a version to keep track of what's in the wild.
+  analytics.SNIPPET_VERSION = '4.1.0';
+
+  // Load Analytics.js with your key, which will automatically
+  // load the tools you've enabled for your account. Boosh!
+  analytics.load('yJpJZWBw68fEj0aPSv8ffMMgof5kFnU9');
+
+  // Make the first page call to load the integrations. If
+  // you'd like to manually name or tag the page, edit or
+  // move this call however you'd like.
+  //
+  // Do not track the first page view, we'll handle this manually
+  //
+  // analytics.page();
+}
+
+export default class SegmentTracker extends BaseTracker {
+  constructor (store) {
+    super()
+
+    this.store = store
+    this.enabled = false
+  }
+
+  async _initializeTracker () {
+    this.store.watch(
+      (state, getters) => getters['me/isTeacher'],
+      this.onIsTeacherChanged.bind(this)
+    )
+
+    if (this.store.getters['me/isTeacher']) {
+      loadSegment()
+      window.analytics.ready(this.onInitializeSuccess)
+    } else {
+      this.onInitializeSuccess()
+    }
+  }
+
+  async trackPageView (includeIntegrations = {}) {
+    await this.initializationComplete
+
+    if (!this.enabled) {
+      return
+    }
+
+    const url = `/${Backbone.history.getFragment()}`
+    window.analytics.page(url, {}, {
+      integrations: {
+        All: false,
+        'Google Analytics': false,
+        'Segment.io': false
+      }
+    })
+  }
+
+  async identify (traits = {}) {
+    await this.initializationComplete
+
+    if (!this.enabled) {
+      return
+    }
+
+    super.identify(traits);
```

Comment:
Is there a rationale or practice for calling this here?

The code that is being called doesn't seem to do anything:
https://github.com/codecombat/codecombat/pull/5776/files#diff-c265a9b66b72604860c9c0fb09dca78bR26

Reply:
This shouldn't be here :).  Good catch!